### ISSUE TITLE
publish compressed file ?

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/node-minecraft-data#185]
it would reduce the size of nmd package on disk
decompression cost at run time would probably be minimal

**Comment by wgaylord** [comment]
minecraft-data now takes up over 145MB of space (149360KB) on disk when installed. A way to easily shrink this size without and extra code would be to minify the JSON data before publishing to nom. (Remove the extra new lines that make it human read-able.)   This alone shrinks it down to around 115MB (116788KB)   Using an actual compression on top of this would be an improvement as well.

**Comment by rom1504** [comment]
Files are already published compressed. Npm does that by default

Is your goal to decrease your local node modules space use ?

**Comment by wgaylord** [comment]
Thats what I assumed this issue was about, was just giving my input.

**Comment by jojomatik** [comment]
> Thats what I assumed this issue was about, was just giving my input.

Yep, that was the original idea of the issue: https://github.com/PrismarineJS/node-minecraft-protocol/issues/978#issuecomment-1094011178

**Comment by rom1504** [comment]
Ah right, forgot I had opened this

**Comment by rom1504** [comment]
https://www.npmjs.com/package/node-stream-zip
+ One big zip should be pretty good here

**Comment by extremeheat** [comment]
Yeah, publishing a zip seems to be best. node-stream-zip doesn't support sync but it seems https://github.com/cthackers/adm-zip does.

If anyone wants to do it the API seems simple to use also.

**Comment by rom1504** [comment]
Another way would be to compress file independently. Kind of same as zip really 

Would just need to try out when to run this, eg pre publish post install etc and see loading time and if that breaks web

**Comment by rom1504** [comment]
tried there https://github.com/PrismarineJS/node-minecraft-data/pull/346
